experimental/rsocket-src/transports/TcpConnectionFactory.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/async/EventBaseManager.h>
+#include <folly/io/async/ScopedEventBaseThread.h>
+#include <rsocket/transports/TcpConnectionFactory.h>
+#include "src/framed/FramedDuplexConnection.h"
+#include "src/tcp/TcpDuplexConnection.h"
+
+using namespace ::reactivesocket;
+using namespace ::folly;
+
+namespace rsocket {
+
+// carry the thread/eventbase with the DuplexConnection
+class FramedDuplexConnectionOnThread : public DuplexConnection {
+ public:
+  FramedDuplexConnectionOnThread(
+      std::unique_ptr<FramedDuplexConnection> fd,
+      std::unique_ptr<ScopedEventBaseThread> eventBaseThread)
+      : fd_(std::move(fd)), eventBaseThread_(std::move(eventBaseThread)) {}
+
+  ~FramedDuplexConnectionOnThread() {
+    LOG(INFO) << "FramedDuplexConnectionOnThread => destroy";
+  }
+
+  void setInput(
+      std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> framesSink) {
+    fd_->setInput(std::move(framesSink));
+  }
+
+  std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> getOutput() {
+    return fd_->getOutput();
+  }
+
+ private:
+  std::unique_ptr<FramedDuplexConnection> fd_;
+  std::unique_ptr<ScopedEventBaseThread> eventBaseThread_;
```

Comment:
I am afraid this might introduce some race errors. When FramedDuplexConnectionOnThread is destroyed, the EventBase is destroyed and at the same time the TCP connection might not be done with the execution/cleanup.
Would it be possible to keep the EventBase outside of the duplex connection? Why do you want to allocate a new thread per tcp connection? You can run all of them on one EventBase, right?